### PR TITLE
[Storage Blobs Service] Blob Tags

### PR DESCRIPTION
**Pull request by Jinming-Hu**

### CODE REVIEW

File: sdk/storage/test/blobs/append_blob_client_test.cpp
Comment (JinmingHu-MSFT) [inline sdk/storage/test/blobs/append_blob_client_test.cpp:116]: https://github.com/Azure/azure-sdk-for-cpp/issues/417

File: sdk/storage/inc/blobs/blob_client.hpp
Comment (katmsft) [inline sdk/storage/inc/blobs/blob_client.hpp:386]: Please fix the format.
To be more specific, any comment should not be split into multiple lines before the collum limit is hit. Also, each new line should be approximately aligned in all multiple-line comments.

File: sdk/storage/inc/blobs/blob_options.hpp
Comment (katmsft) [inline sdk/storage/inc/blobs/blob_options.hpp:31]: Leave a sample or a reference link here please.

**Comment by JinmingHu-MSFT** [comment]
blocked by https://github.com/Azure/azure-sdk-for-cpp/issues/407

File: sdk/storage/inc/blobs/blob_options.hpp
Comment (Jinming-Hu) [reply sdk/storage/inc/blobs/blob_options.hpp]: fixed

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp
Comment (katmsft) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:33]: Please create a work item to do some convenience enhancement to provide functionality for customers to construct the tag condition query string.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp
Comment (katmsft) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:1061]: I don't know if the unnamed struct pattern is a good idea for something that might be reused. Can we bring this to review before we decide that we can write code in this style?

**Review by katmsft — APPROVED** [review]
Approve but there are some pending discussion/items. Also, I think it is important to have a sample for tag's usage as it is not obvious, but it can be added later.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:33]: language alignment is one of our most sought in Track2. Is there any source that suggests they're going to do so in any other language?

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:1061]: We defined separate structs for those that might be reused for example `BlobAccessConditions `. Access conditions for this API is very special. None of any other API has the same set of fields as it. So in this case I think it's fine to use unnamed struct. If in the future, we had a new API that share the same fields, we could define a new struct, it's not even a breaking change.

**Pull request merged**
